src/core/worker.js

```diff
@@ -671,11 +671,9 @@ var WorkerMessageHandler = {
 
       var onFailure = function(e) {
         if (e instanceof PasswordException) {
-          if (e.code === PasswordResponses.NEED_PASSWORD) {
-            handler.send('NeedPassword', e);
-          } else if (e.code === PasswordResponses.INCORRECT_PASSWORD) {
-            handler.send('IncorrectPassword', e);
-          }
+          handler.sendWithPromise('Password', e).then(function (data) {
+            pdfManager.updatePassword(data.password);
+          });
```

Comment:
> add .catch(function (e) { handler.send('PasswordException', e); });. (I'm not sure if we need to re-wrap e, but I think this will work out)

I complication here is that the exception that we will thus `catch`, is actually a `string` rather than an actual `Error` here since `postMessage` has issues with serializing those.

For now, the patch is still sending the original `PasswordException` in this case, since I'm having trouble getting this to work properly otherwise.